### PR TITLE
Add a comment about unix socket authentication with mariadb

### PR DESCRIPTION
**Pull request by toadjaune**
##### ISSUE TYPE
- Docs Pull Request

+label: docsite_pr

##### SUMMARY
Add a comment about unix socket authentication with mariadb

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
mysql_db

##### ADDITIONAL INFORMATION
If this change is accepted, it will make sense to apply it to the other mysql modules.

### CODE REVIEW

**Comment by Andersson007** [comment]
Hey! Is it not about the stuff from the pr https://github.com/ansible/ansible/pull/66848
that has been merged recently?
Anyway if you want to add something and it’s relevant for all the mysql modules, better to use file from my pr above. That’s appended to the documentation of all modules automatically.
needs_info

**Comment by gundalow** [comment]
@toadjaune Hi, what are your thoughts on @Andersson007 comments above?

File: lib/ansible/modules/database/mysql/mysql_user.py [hunk 1]
```diff
@@ -118,6 +118,9 @@
      without providing any login_user/login_password details. The second must drop a ~/.my.cnf file containing
      the new root credentials. Subsequent runs of the playbook will then succeed by reading the new credentials from
      the file."
+   - "When using this module with mariadb, it is possible to work around this problem by using unix socket instead of
+     login/password. In this case, you can disregard the previous note and just use `login_unix_socket: /run/mysqld/mysqld.sock`.
```
Comment (bmalynovytch) [inline lib/ansible/modules/database/mysql/mysql_user.py:122]: ⚠️ Provided path is OS specific

**Review by bmalynovytch — COMMENTED** [review]
My own thought is that this comment should be useless if people using the module knew more about MySQL / MariaDB.
We shouldn't need to give such suggestions as it is more related to how MySQL and MariaDB work than how Ansible's module does.
Maybe would it be best to clarify how Ansible currently is connecting (by default) and warning people about the fact that servers now use other auth methods (by default), such as unix socket, which is out of scope of this documentation, which may lead to "defaults" being different between Ansible and target server.

**Comment by Andersson007** [comment]
@bmalynovytch  i agree with you in case of this pr. Yes, it's a question of understanding mysql.
However if plenty of people stumble over it and complain (as we can see in related issues), it is not a problem to add a small note about a solution to make their lives easier (and i stumbled over it too when i wrote CI tests from scratch for one of mysql modules:).
IMO this PR should be closed because it is already explained by https://github.com/ansible/ansible/pull/66848 in more general way

**Comment by Andersson007** [comment]
@toadjaune we can't accept the PR and the issue because of the reasons mentioned above
close_me

**Comment by toadjaune** [comment]
I agree with most of what was said above, and as far as I'm concerned, #66848 fixes the issue.
Thanks guys :)

**Comment by Andersson007** [comment]
@toajaune no problem, you’re welcome!

**Pull request closed without merging**